### ISSUE TITLE
floating point IO

### ISSUE THREAD

**Issue by mosaicthej** [mosaicthej/qemu-215#3]
Do IO with floating point.

One approach is that let user to move the pattern to `a0`.
Then in the kernel just do printf.

Will try this out.

**Comment by mosaicthej** [comment]
Looked how a simple printf("%f",p) is implemented.
Should be simple enough.... Given the arg is on a0

```asm
	flw	fa0,-20(s0) # bring (p) <- fp-20 onto fa0
	call	__extendsfdf2 # this returns a double (extends it to double)
	mv	a4,a0 # lower half
	mv	a5,a1 # upper half
	mv	a2,a4 # lo
	mv	a3,a5 # hi
	lui	a5,%hi(.LC1)
	addi	a0,a5,%lo(.LC1) # "%f" format str
	# before call: a0 points to str "%f",
	# a2 holds %lo[p] and a3 holds %hi[p]
	call	printf
```

**Comment by mosaicthej** [comment]
closed with 873c27c1

**Comment by mosaicthej** [comment]
closed with 873c27c1
